### ISSUE TITLE
[PyVISA >v1.11 compatibility issues] module write() function expects tuple from pyvisa write(), gets int; custom waveform transfer issue

### ISSUE THREAD

**Issue by laurenwilliams22** [asvela/tektronix-func-gen#2]
I just copied one of the basic examples in the read me file (pasted below for reference). When I run it, I get errors in lines 117 and lines 200 in the tektronix_func_gen.py file. The error is: `TypeError: cannot unpack non-iterable int object`. Do you know why this might be happening? I have pyvisa, NI VISA, and numpy installed. 

```python
import tektronix_func_gen as tfg

with tfg.FuncGen('GPIB0::11::INSTR') as fgen:
      fgen.ch1.set_function("SIN")
      fgen.ch1.set_frequency(25, unit="Hz")
      fgen.ch1.set_offset(50, unit="mV")
      fgen.ch1.set_amplitude(0.002)
      fgen.ch1.set_output("ON")
      fgen.ch2.set_output("OFF")
      # alternatively fgen.ch1.print_settings() to show from one channel only
      fgen.print_settings()
```

**Comment by asvela** [comment]
Hi Lauren,

Just to check, I noticed that in your example code, there are no tabs in front of the lines after the _with_ statement, is this a paste error or do you not have tabs in your code?

This is not something I have seen before. As it is line 117 in _tektronix_func_gen.py_ that's prompting the error at 200, it happens in the initialisation of the device. The error at line 200 suggests that pyvisa is not returning both the number of bytes and the status code. If you temporarily change line 200 to be 
```python
temp = self.inst.write(command)
```
and then insert the lines 
```python
print(type(temp))
print(temp)
```
right after line 200, and run the programme again, what does the terminal show?

**Comment by laurenwilliams22** [comment]
Thank you for your quick reply! I did have the tabs in front of the lines after the with statement, it was a paste error. I inserted those lines and got: 
<class 'int'>
6

**Comment by asvela** [comment]
Okay, I will have to investigate this as it seems the output from the pyvisa library is not as it used to be, but looking through the change logs I can't find anything about it being changed.

However, as a temporary fix, I hope you should be able to run it if you change the write() (lines 200-210) function to be:
```python
num_bytes= self.inst.write(command)
return num_bytes, None
```

Let me know if that solves it for now.

**Comment by laurenwilliams22** [comment]
Thank you so much for your help. I made the changes you suggested and also included the override compatibility as shown below. It works with these changes. Thanks again!

`with tfg.FuncGen('GPIB0::11::INSTR', override_compatibility=True) as fgen:`

**Comment by asvela** [comment]
Good to hear! 

I had a look and discovered that pyvisa has changed a bit with its release of v11, I assume you are using that one? (You can check that by executing `pip list` or `pyvisa-info` in the terminal/command prompt, sorry if I am explaining things you know well, I just find it frustrating when people ask for info but don't tell you how to obtain it.)

I have made some changes to the module now to accommodate for this new pyvisa version. Could you please try using the code on the branch [fix/issue2](https://github.com/asvela/tektronix-func-gen/blob/cfc0f1264c0c6da4466f06762df6b22906e87ef3/tektronix_func_gen.py#L1)?

_PS:_ Out of curiosity, what model of the AFG are you using? The override_compatibility is needed because you are not using AFG1022, but be warned, the code will assume the voltage/frequency limits of the AFG1022:
```
"frequency lims": ({"min": 1e-6, "max": 25e6}, "Hz"),
"voltage lims":   ({"50ohm": {"min": -5, "max": 5},
                    "highZ": {"min": -10, "max": 10}}, "V"),
"amplitude lims": ({"50ohm": {"min": 0.001, "max": 10},
                    "highZ": {"min": 0.002, "max": 20}}, "Vpp")}
```
so if your instrument has larger/smaller limits you will not be able to access the full range/might run into errors.

**Comment by laurenwilliams22** [comment]
Thank you so much for updating the code. I am using v1.11 (I am actually quite a novice at this stuff so honestly the more info the better!). I am using the AFG31102, thanks for the warning about the limits. The new code works perfectly for the original example I was using. When I tried copying the arbitrary waveform example, I ran into pyvisa.errors.VisaIOError: VI_ERROR_TMO (-1073807339): Timeout expired before operation completed. I am using NI VISA. The line in the code that raises the issue is:
` fgen.set_custom_waveform(waveform, memory_num=5, verify=True)`
There still seems to be some issue with pyvisa. I will keep trying to figure this out. Thanks again for all your help.

**Comment by asvela** [comment]
The line in question here was due to a funny problem with previous versions of PyVISA, where an empty query was needed after a 'write_binary_values()'. However, it seems v1.11 solved this and there is no need for the line anymore.

Could you please try commenting out line 479 in your `tektronix_func_gen2.py` (saw you had named it with that extra `2`)? This will hopefully let you run it. 

The AFG31102 is a more advanced instrument than the 1022 that I have tested the package with, so unfortunately your instrument might not be compatible with all the code (the instruments might have slightly different VISA commands for some features). The [programmer's manual](https://download.tek.com/manual/077148802_AFG31000_Series_Programmer_Manual_Sept2019.pdf) for your instrument is then the resource you need – it should be fairly easy to make the necessary changes using the code in the module as a base and make changes as appropriate.

However, all the errors so far have been related to pyvisa, so I hope the two instruments are overlapping enough that all the functionality in this module works for you.

**Comment by asvela** [comment]
@laurenwilliams22 It would be great if commenting out the line I suggested works for you!

**Comment by asvela** [comment]
As far as I know fixed by 9c9a71e, re-open if not.
